extensions/notification/NotificationConfigManager.js

```diff
@@ -1,283 +1,87 @@
-        CONFIG_MANAGER_OPERATION_TYPE,
-    ],
-});
-
-const configGetLag = ZenkoMetrics.createHistogram({
-    name: 's3_notification_config_manager_config_get_seconds',
-    help: 'Time it takes in seconds to get a bucket notification config from MongoDB',
-    labelNames: [
-        CONFIG_MANAGER_CACHE_HIT,
-    ],
-    buckets: [0.001, 0.01, 1, 10, 100, 1000],
-});
-
-const cachedBuckets = ZenkoMetrics.createGauge({
-    name: 's3_notification_config_manager_cached_buckets_count',
-    help: 'Total number of cached buckets in the notification config manager',
-});
-
-function onConfigManagerCacheUpdate(op) {
-    cacheUpdates.inc({
-        [CONFIG_MANAGER_OPERATION_TYPE]: op,
-    });
-    if (op === 'add') {
-        cachedBuckets.inc({});
-    } else if (op === 'delete') {
-        cachedBuckets.dec({});
-    }
-}
-
-function onConfigManagerConfigGet(cacheHit, delay) {
-    configGetLag.observe({
-        [CONFIG_MANAGER_CACHE_HIT]: cacheHit,
-    }, delay);
-}
+const MongoConfigManager = require('./configManager/MongoConfigManager');
+const ZookeeperConfigManager = require('./configManager/ZookeeperConfigManager');
 
 /**
  * @class NotificationConfigManager
  *
- * @classdesc Manages bucket notification configurations, the configurations
- * are directly retrieved from the metastore, and are locally cached. Cache
- * is invalidated using MongoDB change streams.
+ * @classdesc Manages bucket notification configurations
  */
 class NotificationConfigManager {
-    /**
-     * @constructor
-     * @param {Object} params - constructor params
-     * @param {Object} params.mongoConfig - mongoDB config
-     * @param {Logger} params.logger - logger object
-     */
-    constructor(params) {
-        joi.attempt(params, paramsJoi);
-        this._logger = params.logger;
-        this._mongoConfig = params.mongoConfig;
-        this._cachedConfigs = new LRUCache(MAX_CACHED_ENTRIES);
-        this._mongoClient = null;
-        this._metastore = null;
-        this._metastoreChangeStream = null;
-    }
 
-    /**
-     * Connects to MongoDB using the MongoClientInterface
-     * and retreives the metastore collection
-     * @param {Function} cb callback
-     * @returns {undefined}
-     */
-    _setupMongoClient(cb) {
-        const mongoUrl = constructConnectionString(this._mongoConfig);
-        MongoClient.connect(mongoUrl, {
-            replicaSet: this._mongoConfig.replicaSet,
-            useNewUrlParser: true,
-        },
-        (err, client) => {
-            if (err) {
-                this._logger.error('Could not connect to MongoDB', {
-                    method: 'NotificationConfigManager._setupMongoClient',
-                    error: err.message,
-                });
-                return cb(err);
-            }
-            this._logger.debug('Connected to MongoDB', {
-                method: 'NotificationConfigManager._setupMongoClient',
+    constructor(params) {
+        this._usesZookeeperBackend = false;
```

Comment:
seems like an anti-pattern: we already have abstracted away zookeeper/mongo through `_configManagerBackend`, so either
* this needs to be moved to the backend
* every backend needs to have a new method, called at the right time, to trigger this zookeeper-specific handling